### PR TITLE
DR-2814: Adding api route caching and client side fetch call caching

### PR DESCRIPTION
**Pull request by EdwinGuzman**
## Ticket:

- JIRA ticket [DR-2814](https://jira.nypl.org/browse/DR-2814)

## This PR does the following:

- This adds caching for 24hours to the Next.js internal API routes for `/api/featuredItem` and `/api/homepage`.
- These two endpoints are hit by the client through fetch calls. I'm using the `{ cache: "force-cache" }` option which will always use the cache if available unless a hard refresh is made. This is based on the docs but I'm wondering if it is too aggresive. https://developer.mozilla.org/en-US/docs/Web/API/Request/cache
- The [fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) is a browser API but node also supports it. I don't think it's right to aggressively cache the fetch call to Repo API but would it make sense to do so?

## How has this been tested?

Locally.

## Accessibility concerns or updates

- N/A

### Checklist:

- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/components/hero/campaignHero.tsx [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
-import { useEffect, useState } from "react";
 
 const CampaignHero = ({ imageID }) => {
   const [data, setData] = useState<any>({});
 
   useEffect(() => {
     const fetchData = async () => {
-      const response = await fetch(`/api/featuredItem?imageID=${imageID}`);
+      const response = await fetch(`/api/featuredItem?imageID=${imageID}`, {
+        cache: "force-cache",
```
[the remark this answers is not in the document]
Comment (EdwinGuzman) [reply src/components/hero/campaignHero.tsx:14]: Yes, I agree. I was debating whether to use it or not. I will remove it.

[the remark this answers is not in the document]
Comment (EdwinGuzman) [reply src/components/hero/campaignHero.tsx]: Just a note that a hard refresh from the browser does make it a clean fetch.

**Comment by kayleiburke** [comment]
I might be doing something wrong, but I pulled down the latest changes, and I'm getting a new header image every time I refresh the page:
Screenshot 2024-03-07 at 2 17 06 PM

**Comment by EdwinGuzman** [comment]
@kayleiburke that is unrelated but a good thing. That means that the optinmonster script is working -- something that Ali mentioned was not fully working on deployed environments but only locally. It seems that's still true.

**Comment by EdwinGuzman** [comment]
Sorry, I misunderstood. I thought you were talking about the banner.

The Hero displays one image from a set of images. So it _should_ change every time you refresh the page, but if you get image 1234 (for example), whenever you next get image 1234 it should be cached. Since it's random, it could be the 4th or 10th time you refresh. That does make this difficult to test.

**Review by chrisb87 — APPROVED** [review]

**Review by kayleiburke — APPROVED** [review]

**Review by avertrees — APPROVED** [review]

**Pull request merged**